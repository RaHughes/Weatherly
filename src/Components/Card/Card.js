import React from 'react';
import './Card.scss'

const Card = ({data}) => {
const cards = data.map(card => {

  return(
    <div key={Math.random()} className='card'>
    <p>Date: {card.dt_txt}</p>
    <img src={`http://openweathermap.org/img/wn/${card.weather[0].icon}@2x.png`}/>
    <p>High/Low: {((card.main.temp_max - 273.15) * 9/5 + 32).toFixed(2)} F / {((card.main.temp_min - 273.15) * 9/5 + 32).toFixed(2)} F </p>
    </div>
  )
})
  
return (
  <section className='cards'>
    {cards}
  </section>
)

}

export default Card;