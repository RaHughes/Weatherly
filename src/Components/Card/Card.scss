.cards {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  min-height: 100vh;
  
  .card {
    justify-content: space-between;
    height: 200px;
    width: 400px;
    border: 2px solid black;
    margin: 10%;
    background-color: lightgray;
    &:hover {
      box-shadow: 0 0 10px blue;
      transition: 0.4s;

    }
      img {
      border: 2px solid gray;
      border-radius: 50%;
      background-color: gray;
    }
  }
}

