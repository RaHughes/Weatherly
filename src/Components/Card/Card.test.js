import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('Should match the snapshot of the DOM', () => {
    const mockData = [{ 
      dt_txt: 'date', 
      weather: [{icon: 'img'}], 
      main: { temp_max: 250, temp_min: 240 }
      }]
      
    const wrapper = shallow(<Card data={mockData}/>)
    expect(wrapper).toMatchSnapshot();
  })
})