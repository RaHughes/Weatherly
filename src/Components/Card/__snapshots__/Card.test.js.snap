// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match the snapshot of the DOM 1`] = `ShallowWrapper {}`;
